eclipse-scout-core/src/table/TableHeader.ts

```diff
@@ -310,7 +311,11 @@ export class TableHeader extends Widget implements TableHeaderModel {
       return column.headerTooltipText;
     }
     let $text = $col.children('.table-header-item-text');
-    if ($text.isContentTruncated() || ($col.width() + $col.position().left) > $col.parent().width()) {
+    let textBounds = graphics.offsetBounds($text);
+    let containerBounds = this._offsetBoundsWithoutMenuBar();
+    let textInView = textBounds.right() < containerBounds.right() && textBounds.x >= containerBounds.x;
+    if ($text.isContentTruncated() || !textInView) {
```

Comment:
The 3 lines above are always executed, even when the first "if" clause returns already true. Either extract them into a _isTextInView() method, swap the if clauses (we can then save the isContentTruncated() call if the text is not in view), or combine them into a "showTooltip" variable.